desc/magnetic_fields.py

```diff
@@ -1637,6 +1644,351 @@ def from_surface(
+        self._shift_basis = ChebyshevDoubleFourierBasis(
+            L=self.L_shift,
+            M=self.M_shift,
+            N=self.N_shift,
+            NFP=self.NFP,
+            sym="cos(t)",
+        )
+        if B_lm is None:
+            self._B_lm = np.concatenate(
+                (
+                    np.ones((self.M_well,)),
+                    np.zeros((self.L_well * self.M_well,)),
+                )
+            )
+        else:
+            assert len(B_lm) == (self.L_well + 1) * self.M_well
+            self._B_lm = B_lm
+        if x_lmn is None:
+            self._x_lmn = np.zeros(self.shift_basis.num_modes)
+        else:
+            assert len(x_lmn) == self.shift_basis.num_modes
+            self._x_lmn = x_lmn
+
+        warnif(
+            self.helicity != (0, self.NFP) and self.helicity[0] != 1,
+            UserWarning,
+            "Typical helicity (M,N) has M=1.",
+        )
+        warnif(
+            self.helicity != (1, 0) and self.helicity[1] != self.NFP,
+            UserWarning,
+            "Typical helicity (M,N) has N=NFP.",
+        )
+
+    def change_resolution(
+        self,
+        L_well=None,
+        M_well=None,
+        L_shift=None,
+        M_shift=None,
+        N_shift=None,
+        NFP=None,
+    ):
+        """Set the spectral resolution of well and shift terms.
+
+        Parameters
+        ----------
+        L_well : int
+            Radial resolution of the magnetic well parameters B_lm.
+        M_well : int
+            Number of spline points in the magnetic well parameters B_lm.
+        L_shift : int
+            Radial resolution of x_lmn.
+        M_shift : int
+            Poloidal resolution of x_lmn.
+        N_shift : int
+            Toroidal resolution of x_lmn.
+        NFP : int
+            Number of field periods.
+
+        """
+        self._NFP = setdefault(NFP, self.NFP)
+        self._L_well = setdefault(L_well, self.L_well)
+        self._M_well = setdefault(M_well, self.M_well)
+        self._L_shift = setdefault(L_shift, self.L_shift)
+        self._M_shift = setdefault(M_shift, self.M_shift)
+        self._N_shift = setdefault(N_shift, self.N_shift)
+
+        old_modes_well = self.well_basis.modes
+        old_modes_shift = self.shift_basis.modes
+
+        self.well_basis.change_resolution(self.L_well)
+        self.shift_basis.change_resolution(
+            self.L, self.M, self.N, NFP=self.NFP, sym="cos(t)"
+        )
+
+        self._x_lmn = copy_coeffs(self.x_lmn, old_modes_shift, self.shift_basis.modes)
+
+        old_well = self.B_lm.reshape((-1, self.M_well))
+        new_well = np.zeros((self.L_well + 1, self.M_well))
+        for j in range(self.M_well):
+            new_well[:, j] = copy_coeffs(
+                old_well[:, j], old_modes_well, self.well_basis.modes
+            )
+        self._B_lm = new_well.flatten()
+
+    # TODO: reuse compute function?
+    def compute_well(self, rho, eta):
```

Comment:
does this need to be a standalone method? or can it be part of regular `compute` api?

Reply:
This was removed and made part of the regular compute API